### ISSUE TITLE
Problem with hanging node position

### ISSUE THREAD

**Issue by SiqiZhang** [geodynamics/aspect#283]
Hi all,

After update to the development version, I found the the hanging node position in output is separated from the line it sits on. I wonder if is there a bug in the mapping of hanging node, or it's something wrong with the dependent packages I built.
Here is the main packages version I used:
P4EST   1.1
Trilinos  11.12.1
Deal.II    8.2.1

To repeat this problem:
Use tests/simple-incompressible.prm
Change two lines in "subsection Mesh refinement" to get some refinement :
set Initial adaptive refinement        = 3
set Strategy                                    = topography

Would anyone give it a try see if it is repeatable?

Regards,

Siqi

![hanging_node_error](https://cloud.githubusercontent.com/assets/7585187/6384731/7ad06786-bdb4-11e4-926b-ab8f86e341b7.png)

**Comment by jperryhouts** [comment]
I suspect this is just paraview incorrectly interpolating the edges after the changes to the shell geometry model in PR #242. If that's the case, it would be nice to figure out a way to make it do so, because that's going to look pretty bad in figures.

**Comment by tjhei** [comment]
Yes, this is due to the manifold changes. The computation is correct, it is just that the graphical output only outputs straight lines (we can not change that). Imagine the longer edge is curved in the computation.

I am not sure how to fix this easily. We could move hanging nodes onto the parent face in the graphical output? @bangerth ?

**Comment by bangerth** [comment]
Actually, it's a bit more complicated. First, it is true that the visualization does not accurately reflect the reality of the simulation. In ASPECT, every edge is a quartic polynomial, not a straight line (a linear polynomial). However, visualization file formats do not allow to represent this, and consequently visualization programs do not show it. This is at a fundamental level the same issue as discussed in this entry of the deal.II FAQs: https://github.com/dealii/dealii/wiki/Frequently-Asked-Questions#in-my-graphical-output-the-solution-appears-discontinuous-at-hanging-nodes

But, there is also the issue that when you have a hanging node, the quartic polynomial line that represents the long edge goes through points at 1/4, 1/2 and 3/4 of the arc that are on the same radius as the end points. Consequently, it does go through the hanging node. On the other hand, the one quartic line for the long edge does not exactly coincide with the two quartic lines for the two short edges. There is indeed a small gap or overlap between the big and the two adjacent small cells. This does introduce an error in the computations, but it is proportional to the area of the gap/overlap and consequently is O(h^5) for a fourth order polynomial. This is smaller than the discretization error, and consequently is a problem that is commonly ignored.


So, to sum up, there are two problems: A visualization problem we cannot do anything about, and a geometric problem that introduces an error that is negligible.
